spark-jobs/src/main/scala/za/co/absa/enceladus/standardization/StandardizationReader.scala

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright 2018 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.enceladus.standardization
+
+import org.apache.spark.sql.{DataFrameReader, SparkSession}
+import org.slf4j.Logger
+import za.co.absa.enceladus.common._
+import za.co.absa.enceladus.dao.MenasDAO
+import za.co.absa.enceladus.model.Dataset
+import za.co.absa.enceladus.utils.unicode.ParameterConversion._
+
+import scala.collection.immutable.HashMap
+
+class StandardizationReader(log: Logger) {
+  private final val SparkCSVReaderMaxColumnsDefault: Int = 20480
+
+  /**
+   * Returns a Spark reader with all format-specific options applied.
+   * Options are provided by command line parameters.
+   *
+   * @param cmd             Command line parameters containing format-specific options
+   * @param dataset         A dataset definition
+   * @param numberOfColumns (Optional) number of columns, enables reading CSV files with the number of columns
+   *                        larger than Spark default
```

Comment:
I think implicit should be named as well and carefully described where they should come from